### PR TITLE
Modernize Course Instance Settings page

### PR DESCRIPTION
**Pull request by mylesw27**
This PR intends to convert the `instructorInstanceAdminSettings` page from JavaScript and EJS to TypeScript and HTML template literals.

### CODE REVIEW

File: apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.html.ts [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+                        >
+                          <i class="fas fa-qrcode"></i>
+                        </button>
+                      </div>
+                    </span>
+                    <div class="d-none">
+                      <div id="js-student-link-qrcode">
+                        <center>${resLocals.studentLinkQRCode}</center>
+                      </div>
+                    </div>
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.html.ts:157]: This doesn't work correctly:

Screenshot 2024-04-02 at 09 58 42

You probably need an `unsafeHtml(...)` wrapping the `resLocals.studentLinkQRCode` to ensure that it doesn't get escaped.

File: apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.html.ts [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    <div
+      class="modal fade"
+      id="deleteCourseInstanceModal"
+      tabindex="-1"
+      role="dialog"
+      aria-labelledby="deleteCourseInstanceModalLabel"
+    >
+      <div class="modal-dialog" role="document">
+        <div class="modal-content">
+          <div class="modal-header">
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.html.ts]: Using a `Modal` component in [9257025](https://github.com/PrairieLearn/PrairieLearn/pull/9673/commits/9257025e843a376237a056af7c027acc5002c911).

File: apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.html.ts (hunk 1 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.html.ts]: Added `unsafeHtml(...)` here and QR code is now working in [9257025](https://github.com/PrairieLearn/PrairieLearn/pull/9673/commits/9257025e843a376237a056af7c027acc5002c911).

File: apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.html.ts [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+        class="btn btn-sm btn-primary"
+        href="#"
+        data-toggle="modal"
+        data-target="#deleteCourseInstanceModal"
+      >
+        <i class="fa fa-times" aria-hidden="true"></i> Delete this course instance
+      </button>
+    </div>
+    ${Modal({
+      id: 'delete-course-instance-form',
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.html.ts:253]: It seems like you didn't test this after changing it? This doesn't match the `data-target` used above (`#deleteCourseInstanceModal`) so the modal will never open.

```suggestion
      id: 'deleteCourseInstanceModal',
```

You'll have to adjust the tests too. Note that this ID is not attached to a `<form>`, so we should not include `-form` in the name.

Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.html.ts]: Updated this as well as the tests in [b354de3](https://github.com/PrairieLearn/PrairieLearn/pull/9673/commits/b354de348fa4ec5559c6b0a4b48afd1e7ca76e73). 

We had used the other naming syntax (`delete-question-form`) when implementing a similar modal for deleting questions. I will update the ID of that one in subsequent PR to match this.

**Pull request merged**